src/functions.jl

```diff
@@ -6,649 +6,646 @@
 # MIT Expat license
 #
 
-## Exp ## 
-doc"""
-    exp(a)
-
-Return the Taylor expansion of $e^a$, of order `a.order`, for
-`a::Taylor1` polynomial.
-
-For details on making the Taylor expansion, see
-[`TaylorSeries.expHomogCoef`](@ref).
-"""
-function exp(a::Taylor1)
-    @inbounds aux = exp( a[1] )
-    T = typeof(aux)
-    v = convert(Array{T,1}, a.coeffs)
-    coeffs = similar(v)
-    @inbounds coeffs[1] = aux
-    @inbounds for k = 1:a.order
-        coeffs[k+1] = expHomogCoef(k, v, coeffs)
-    end
-    Taylor1( coeffs, a.order )
-end
-
-# Homogeneous coefficients for exp
-doc"""
-    expHomogCoef(kcoef, ac, coeffs)
-
-Compute the `k-th` expansion coefficient of $c = \exp(a)$ given by
-
-\begin{equation*}
-c_k = \frac{1}{k} \sum_{j=0}^{k-1} (k-j) a_{k-j} c_j,
-\end{equation*}
 
-with $a$ a `Taylor1` polynomial.
-
-Inputs are the `kcoef`-th coefficient, the vector of the expansion coefficients
-`ac` of $a$, and the already calculated expansion coefficients `coeffs` of `c`.
-"""
-function expHomogCoef{T<:Number}(kcoef::Int, ac::Array{T,1}, coeffs::Array{T,1})
-    kcoef == 0 && return exp(ac[1])
-    coefhomog = zero(T)
-    @inbounds for i = 0:kcoef-1
-        coefhomog += (kcoef-i) * ac[kcoef-i+1] * coeffs[i+1]
-    end
-    coefhomog = coefhomog/kcoef
-    coefhomog
-end
+# Functions
+for T in (:Taylor1, :TaylorN)
+    @eval begin
+        ## exp ##
+        function exp(a::$T)
+            if $T == Taylor1
+                order = a.order
+                @inbounds aux = exp( a[1] )
```

Comment:
I just noticed that, here, I made a design decision. Suppose that the maximum order (for `TaylorN`) is 6. By what ever reason, you may have the (first) independent variable of order 2, `x=TaylorN(1, order=2)`; yet, `exp(x)` should be returned to maximum possible order (so `get_order()`, instead of `get_order(x) = x.order`...